services/bundle_analysis.py

```diff
@@ -109,8 +109,8 @@ def __init__(self, report: SharedBundleAnalysisReport):
         self.cleanup()
 
     def cleanup(self) -> None:
```

Comment:
Don't do the full `cleanup` since that will delete the DB temp file prematurely.

Reply:
Do we still need the cleanup at this level? I feel having a cleanup here might be more confusing than not, since all connection related stuff should be done at the parent (shared) level

Reply:
We still need this for now, but you're right it shouldn't have to do this on this level. Reason being on the `shared.BundleAnalysisReport` it is still relying on the implementation of manually closing connections. I didn't want to touch that in the current shared PR because `BundleAnalysisReport` is too intertwined with worker as well. But yeah next step I need to make some changes to `BundleAnalysisReport` class